packages/platform-ios/src/commands/runIOS/index.ts

```diff
@@ -98,20 +115,48 @@ function runIOS(_: Array<string>, ctx: Config, args: FlagsT) {
         )}". ${printFoundDevices(devices)}`,
       );
     }
+
     if (device.type === 'simulator') {
       return runOnSimulator(xcodeProject, scheme, args);
-    } else {
-      return runOnDevice(device, scheme, xcodeProject, args);
-    }
-  } else {
-    const physicalDevices = devices.filter(d => d.type !== 'simulator');
-    const device = matchingDevice(physicalDevices, args.device);
-    if (device) {
-      return runOnDevice(device, scheme, xcodeProject, args);
     }
+
+    return runOnDevice(device, scheme, xcodeProject, args);
+  }
+
+  const physicalDevices = devices.filter(d => d.type !== 'simulator');
+  const device = matchingDevice(physicalDevices, args.device);
+  if (device) {
+    return runOnDevice(device, scheme, xcodeProject, args);
   }
 }
 
+async function promptWhichDeviceToRun(devices: Device[]): Promise<Device> {
```

Comment:
How about moving this to `utils` so that both iOS/Android (and possibly other platforms in the future) can take advantage of that?